airflow/api_connexion/endpoints/plugin_endpoint.py

```diff
@@ -0,0 +1,33 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from airflow.api_connexion import security
+from airflow.api_connexion.exceptions import NotFound
+from airflow.api_connexion.schemas.plugin_schema import PluginCollection, plugin_collection_schema
+from airflow.plugins_manager import get_plugin_info
+from airflow.security import permissions
+
+
+@security.requires_access([(permissions.ACTION_CAN_READ, permissions.RESOURCE_PLUGIN)])
+def get_plugins():
+    """Get plugins endpoint"""
+    plugins_info = get_plugin_info()
+    if plugins_info:
+        return plugin_collection_schema.dump(
+            PluginCollection(plugins=plugins_info, total_entries=len(plugins_info))
+        )
+    raise NotFound(detail="Plugin not found")
```

Comment:
Should this return an empty (but successful) response rather than a 404 if there are no plugins? I believe this is the behaviour of other `list` endpoints.

Reply:
oh yeah, I think so -- we should just return an empty list as we don in the Webserver:

https://github.com/apache/airflow/blob/25fa3092323e0cd13b11bb8e809b88ff3c043e4e/airflow/www/templates/airflow/plugin.html#L28-L31

Reply:
I'm also thinking if we should have `limit` and `offset` on this endpoint? That's pagination. @kaxil @mik-laj

Reply:
Yes. Yes.  we should add these parameters if it is possible to have a unified interface which will allow dynamically generating clients and applications based on Opec API spec.

Reply:
Yup, we should